cxx/include/mspass/utility/MetadataDefinitions.h

```diff
@@ -104,20 +105,147 @@ class MetadataDefinitions
+
+  In MongoDB a normalized attribute is one that has a master copy in one and
+  only one place.   This method returns true if an attribute is marked
+  normalized and false otherwise (It will also return false for any key that
+  is undefined.).
+  */
+  bool is_normalized(const string key) const;
+  /*! \brief Returns a unique identifier for a normalized attribute.
+
+  In MongoDB a normalized attribute is one that has a master copy in one and
+  only one place.  This method returns a unique identifier, which defines
+  a key that can be used to access the attribute used as an index to
+  identify a unique location for an otherwise potentially ambiguous identifier
+  (e.g. sta can be used in may contexts).  The type of attribute to which the
+  returned key is linked is expected to normally be acquired by am immediate
+  call to the type method of this class using the return key.  It is the callers
+  responsibility to handle errors if the request for the type information fails.
+  Note for MongoDB the most common (and recommended) type for the unique id is
+  an Object_ID.   The design of the API, however, should not preclude some other
+  index or an index oriented toward a relational database.  (e.g. chanid
+  is an integer key with a one-to-one relation for channel data in the CSS3.0
+  schema.)
+
+  Some unique id specifications require a table/document qualifier.   See
+  related table method that is designed to handle that.
+
+  This method should normally be used only on read operations
+  to select the correct entry for what could otherwise be a potentially ambiguous key.
+  \param key is the flat namespace key for which normalizing data is needed
+
+  \return  name for unique id for requested key.  Returns an empty string if
+    the key is not defined as normalized.   In multiple calls it is more efficient
+    to test for a null return and handle such entries inline instead of a double
+    search required if preceded by is_normalized.
+  */
+  string unique_id_key(const string key) const;
+  /*! \Brief return the master table for a key used as a unique id.
+
+  Support for normalized Metadata requires static tables (collection in MongoDB)
+  that contain the data using normalization.   In seismic data type examples are
+  receiver location tables, receiver response tables, and source location data.
+  This method should nearly always be paired with a call to unique_id_key.
+  The idea is to first ask for the unique_id_key and then ask what table (collection)
+  contains the key returned by unique_id_key.   This provides a fast and
+  convenient lookup for normalized data.
+
+  \param key is the normally the return from unique_id_key
+  \return string defining the table(collection) the key can be used for locating the
+     unique tuple/document required to access related Metadata.  String will be
+     empty if the search fails.
+  */
+  string table(const string key) const;
```

Comment:
Tha `table` is a relational database concept, and the `normalize` above is a MongoDB thing. This could be confusing to the users. I think we should either use `collection` & `normalize` or `table` & `join` so the concepts are consistent. I guess the question is whether we want to hide the new concepts of MongoDB with the more familiar relational ones.

Reply:
Excellent point.  Easily fixed.  Will do so immediately and push the changes.